File: PRAK403-2310817210019-NaufalElyzar.py
AngkaPertama, AngkaKedua = input().split()
AngkaPertama = int (AngkaPertama)
AngkaKedua = int (AngkaKedua)
x = AngkaPertama
y = AngkaKedua
print (f"{AngkaPertama} {AngkaKedua}", end= " - ")
if AngkaPertama > AngkaKedua :
    jarak = AngkaPertama - AngkaKedua
    for i in range (jarak - 1) :
        AngkaPertama -=1
        AngkaKedua +=1
        print (f" {AngkaPertama} {AngkaKedua}", end= " - ")
elif AngkaPertama < AngkaKedua :
    jarak = AngkaKedua - AngkaPertama
    for i in range (jarak - 1) :
        AngkaPertama +=1
        AngkaKedua -=1
        print (f"{AngkaPertama} {AngkaKedua}", end= " - ")
print (f"{y} {x}")

File: PRAK403-2310817210019-NaufalElyzar.c
#include <stdio.h>

int main() {
    int AngkaPertama, AngkaKedua, i, j, x, y;
    scanf("%d %d", &AngkaPertama, &AngkaKedua);
    x = AngkaPertama;
    y = AngkaKedua;

    if (AngkaPertama > AngkaKedua) {
        for(i = AngkaPertama, j = AngkaKedua; i >= AngkaKedua, j <= AngkaPertama; i--, j++) {
                while(AngkaPertama >= AngkaKedua){
                printf("%d %d - ", AngkaPertama, AngkaKedua);
                AngkaPertama--;
                AngkaKedua++;
                continue;
            }
            printf("%d %d - ", AngkaPertama, AngkaKedua);
            AngkaPertama--;
            AngkaKedua++;
            if(AngkaPertama == y){
                    printf("%d %d", AngkaPertama, AngkaKedua);
            }
        }
    }
    else {
        for(i = AngkaPertama, j = AngkaKedua; i <= AngkaKedua, j >= AngkaPertama; i++, j--) {
                while(AngkaPertama <= AngkaKedua){
                printf("%d %d - ", AngkaPertama, AngkaKedua);
                AngkaPertama++;
                AngkaKedua--;
                continue;
            }
            printf("%d %d - ", AngkaPertama, AngkaKedua);
            AngkaPertama++;
            AngkaKedua--;
            if(AngkaKedua == x){
                    printf("%d %d", AngkaPertama, AngkaKedua);
            }
        }
    }
    return 0;
}